### PR TITLE
Genre class

### PR DESCRIPTION
**Pull request by Goruchie**
# **Add the following features to the project:**
- **Create the class Genre that provides the arguments**

**1. id: int
2. name: string
3. items: Arra**

- **Adding the add_item  method at this moment is not possible because we need the Item class first, but when that is ready we can complete this required method for adding items.**

### CODE REVIEW

File: genre.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './item'
```
Comment (Haywayaheadshot) [inline genre.rb:1]: Please remove or comment out this line as it would throw errors when you run the unit tests

File: genre.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+class Genre
+  attr_accessor :name, :items
+
+  def initialize(name, _items)
+    @id = Random.rand(1..1000)
+    @name = name
+    @items = []
+  end
+end
```
Comment (Haywayaheadshot) [inline genre.rb:11]: Please add an add_item method to the Genre class as required by the project. The add_item method should,
- take an instance of the Item class as an input.
- add the input item to the collection of items.
- add self as a property of the item object (by using the - correct setter from the item object).

Please check the [requirement](https://github.com/users/Haywayaheadshot/projects/7/views/1?pane=issue&itemId=18112842)

Comment (Haywayaheadshot) [inline genre.rb:11]: Please add unit tests for all methods in the class.

**Review by Haywayaheadshot — CHANGES_REQUESTED** [review]
## Hi Goruchie, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
- [ ] Please check the changes left as comments.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by StarMindz — APPROVED** [review]

**Review by Haywayaheadshot — APPROVED** [review]

**Review by StarMindz — APPROVED** [review]
Nice job 🙌
[](https://media.giphy.com/media/5hgYDDh5oqbmE4OKJ3/giphy.gif)

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

Highlights
✔️ Descriptive Readme
✔️ Passing Linters
✔️ Correct Gitflow

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**